packages/@aws-cdk/aws-batch/lib/compute-environment.ts

```diff
@@ -0,0 +1,432 @@
+     *
+     * If specified, and this is an managed compute environment, the property will be ignored.
+     *
+     * By default, AWS Batch managed compute environments use a recent, approved version of the
+     * Amazon ECS-optimized AMI for compute resources.
+     *
+     * @default - AWS-managed compute resources
+     */
+    readonly computeResources?: ComputeResourceProps;
+
+    /**
+     * The state of the compute environment. If the state is set to true, then the compute
+     * environment accepts jobs from a queue and can scale out automatically based on queues.
+     *
+     * @default ComputeEnvironmentStatus.ENABLED
+     */
+    readonly state?: ComputeEnvironmentStatus;
+
+    /**
+     * Whether the compute resources should scale automatically based on job queues.
+     *
+     * @default MANAGED
+     */
+    readonly type?: ComputeEnvironmentType;
+}
+
+/**
+ * Properties of a compute environment
+ */
+export interface IComputeEnvironment extends IResource {
+    /**
+     * The ARN of this compute environment
+     *
+     * @attribute
+     */
+    readonly computeEnvironmentArn: string;
+
+    /**
+     * The name of this compute environment
+     *
+     * @attribute
+     */
+    readonly computeEnvironmentName: string;
+}
+
+/**
+ * Batch Compute Environment
+ *
+ * Defines a batch compute environment to run batch jobs on.
+ */
+export class ComputeEnvironment extends Resource implements IComputeEnvironment {
+    /**
+     * Fetches an existing batch compute environment by its amazon resource name.
+     *
+     * @param scope
+     * @param id
+     * @param computeEnvironmentArn
+     */
+    public static fromComputeEnvironmentArn(scope: Construct, id: string, computeEnvironmentArn: string): IComputeEnvironment {
+        const stack = Stack.of(scope);
+        const computeEnvironmentName = stack.parseArn(computeEnvironmentArn).resource;
+
+        class Import extends Resource implements IComputeEnvironment {
+            public readonly computeEnvironmentArn = computeEnvironmentArn;
+            public readonly computeEnvironmentName = computeEnvironmentName;
+        }
+
+        return new Import(scope, id);
+    }
+
+    /**
+     * The ARN of this compute environment
+     *
+     * @attribute
+     */
+    public readonly computeEnvironmentArn: string;
+
+    /**
+     * The name of this compute environment
+     *
+     * @attribute
+     */
+    public readonly computeEnvironmentName: string;
+
+    constructor(scope: Construct, id: string, props: ComputeEnvironmentProps = {}) {
+        super(scope, id, {
+            physicalName: props.computeEnvironmentName,
+        });
+
+        this.validateProps(props);
+
+        let computeResources: CfnComputeEnvironment.ComputeResourcesProperty | undefined;
+
+        // Only allow compute resources to be set when using MANAGED type
+        if (props.computeResources && props.type === ComputeEnvironmentType.UNMANAGED) {
```

Comment:
Yep, the comment was definitely wrong here.

It's a good idea. I am going to add checks to make sure that based on `type`, we have expectations for compute resources being provided. Should we delve deeper and throw errors for setting spot properties (i.e. `spotIamFleetRole`) when the `type` is `ON_DEMAND` ?

Reply:
> Should we delve deeper and throw errors for setting spot properties (i.e. `spotIamFleetRole`) when the type is `ON_DEMAND`?

That I would probably skip for now - doesn't seem worth it (perhaps experience using the library will teach us otherwise, but I wouldn't add it in this PR).